lib-python/littleboxes/solver/cluedb_solver.py

```diff
@@ -1,15 +1,225 @@
 from littleboxes.solver.solver import Solver
+from littleboxes.xword import InvalidCrosswordException
+
+from copy import deepcopy
+import itertools
+import time
 
 class ClueDBSolver(Solver):
     """Solver that looks for clue matches in the provided database."""
+
     def __init__(self, db, distance_cutoff=0):
         """Args:
             db (ClueDB): The database of clues to search for answers.
-            distance_cutoff (int): Clues will only be considered a match if they
-                have Levenstein distance <= cutoff. Default is 0 meaning exact match.
+            distance_cutoff (int): Clues will only be considered a match if 
+                they have Levenstein distance <= cutoff. Default is 0 meaning
+                exact match.
         """
         self._db = db
         self._distance_cutoff = distance_cutoff
 
     def solve(self, xword):
-        pass
+        '''Set-based search for partial solutions to a crossword
+
+        This implementation is highly speed-efficiently, and very memory
+        efficient (although there could be improvements to the latter by not
+        loading certain things into memory at once)
+
+        Precomputes possible answers to every clue using query_answers and
+        determines what conflicts exist between across and down answers first
+
+        Arguments:
+            xword - a Crossword to be solved
+
+        Returns:
+            sorted list of tuples (Crossword, prob) of partially solved 
+            Crosswords in order from most- to least-solved with prob=1.0 for
+            each Crossword for now
+        '''
+
+        possible_answers = self.query_answers(xword)
+
+        start_conflicts = time.time()
+        conflict_map = self.build_conflict_map(xword, possible_answers)
+
+        print 'Conflicts map build in {0} seconds'.format(
```

Comment:
I didn't realize it was a convention, but that's exactly what I did for senior design
![screen shot 2016-02-27 at 6 17 20 pm](https://cloud.githubusercontent.com/assets/9629191/13376051/697273ae-dd7e-11e5-8e17-cc225ea64bda.png)

So I like it, I'll do that